src/jetstream.rs

```diff
@@ -1050,175 +1086,157 @@ impl JetStream {
     }
 }
 
-/// `JetStream` reliable consumption functionality.
-pub struct Consumer {
-    /// The underlying NATS client
-    pub js: JetStream,
+#[derive(Debug)]
+pub(crate) enum ConsumerKind {
+    Ephemeral,
+    Persistent,
```

Comment:
`Durable` is more NATS-idiomatic way to call it instead of Persistent.

Reply:
Does it have to be durable tho? Just meant to communicate internally if we bound to an existing consumer or created an ephemeral one.

Reply:
Renamed this to a simple but still explicit `ConsumerOwnership` boolean enum (`Yes` or `No`).

Reply:
I'll go ahead and mark this as resolved.